File: abc195/src/bin/b.py
a,b,w = map(int,input().split())

def possible(i):
    x = w * 1000 / i
    return a <= x and x <= b

ns = [i for i in range(1, 1000001) if possible(i)]
if len(ns) > 0:
    print(min(ns), max(ns))
else:
    print("UNSATISFIABLE")
"""
min(ns) = m
W = w * 1000
W / m <= b
max(ns) = M
W / M >= a  <=> W >= a * M
y = W // a 
"""

File: abc195/src/bin/b.rs
fn ceil(a: usize, b: usize) -> usize {
    (a + (b - 1)) / b
}
#[test]
fn test_ceil() {
    assert_eq!(ceil(7,3), 3);
}
fn main() {
    proconio::input! {
	a: usize,
	b: usize,
	w: usize,
    }
    let w = w * 1000;
    let ns = || (1..=1_000_000)
	.filter(|&i| {
	    let w = w as f64 / i as f64;
	    a as f64 <= w && w <= b as f64
	});
    if let Some(_) = ns().min() {
	println!("{} {}", ns().min().unwrap(), ns().max().unwrap());
    } else {
	println!("UNSATISFIABLE");
    }
}
